stdlib/REPL/src/REPLCompletions.jl

```diff
@@ -213,8 +214,26 @@ function complete_symbol(sym::String, @nospecialize(ffunc), context_module::Modu
                 end
             end
         end
+
+        # Also, try abstract-interpreting propertynames
+        thunk = context_module.eval(:(() -> propertynames($(ex))))
+        code_info, rett = Core.Compiler.typeinf_code(
+            Core.Compiler.NativeInterpreter(),
```

Comment:
For this to be meaningfully powerful, i think you probably need an interpreter with `assume_bindings_static` turned on.

Reply:
We should separate the cache for this inference that is much aggressive than the ordinary inference.